### PR TITLE
Implement bottom up validation

### LINKED ISSUE #345: Implement bottom up path validation
The current code does little path validation since the required identifiers recently changed in #334 

This should be implemented as @AdityaSripal specifies in this [comment](https://github.com/cosmos/relayer/issues/329#issuecomment-733705336)

### PR DESCRIPTION
**Pull request by akhilkumarpilli**
Fixes: #345

### CODE REVIEW

File: cmd/config.go [hunk 1]
```diff
@@ -417,3 +419,129 @@ func overWriteConfig(cmd *cobra.Command, cfg *Config) error {
...
+		return err
+	}
+	if p.Src.Order != p.Dst.Order {
+		return fmt.Errorf("both sides must have same order ('ORDERED' or 'UNORDERED'), got src(%s) and dst(%s)",
+			p.Src.Order, p.Dst.Order)
+	}
+	return nil
+}
+
+// ValidatePathEnd allow validates provided pathend and returns error for invalid identifiers
```
Comment (anilcse) [inline cmd/config.go]:
```suggestion
// ValidatePathEnd validates provided pathend and returns error for invalid identifiers
```

File: cmd/config.go [hunk 2]
```diff
@@ -417,3 +419,129 @@ func overWriteConfig(cmd *cobra.Command, cfg *Config) error {
...
+
+	chain, err := c.Chains.Get(pe.ChainID)
+	if err != nil {
+		return err
+	}
+
+	height, err := chain.QueryLatestHeight()
+	if err != nil {
+		return err
+	}
```
Comment (anilcse) [inline cmd/config.go]: This is duplicted in multiple places. May be better to change `ValidateClient` as `ValidateClient(chain, height, pathEnd)` . Also for `ValidateConnection` and `ValidateChannel`

**Review by anilcse — CHANGES_REQUESTED** [review]

**Review by colin-axner — APPROVED** [review]
LGTM, great work!

File: cmd/config.go [hunk 3]
```diff
@@ -417,3 +419,110 @@ func overWriteConfig(cmd *cobra.Command, cfg *Config) error {
...
 }
+
+// ValidatePath checks that a path is valid
+func (c *Config) ValidatePath(p *relayer.Path) (err error) {
+	if err = c.ValidatePathEnd(p.Src); err != nil {
+		return err
+	}
+	if p.Src.Version == "" {
+		return fmt.Errorf("source must specify a version")
+	}
```
Comment (anilcse) [inline cmd/config.go]: Let's re-order this check
```suggestion
	if p.Src.Version == "" {
		return fmt.Errorf("source must specify a version")
	}
	if err = c.ValidatePathEnd(p.Src); err != nil {
		return err
	}
```

File: cmd/config.go [hunk 4]
```diff
@@ -417,3 +419,110 @@ func overWriteConfig(cmd *cobra.Command, cfg *Config) error {
...
+	if pe.ConnectionID != "" {
+		if err := c.ValidateConnection(chain, height, pe); err != nil {
+			return err
+		}
+	}
+	if pe.ChannelID != "" {
+		if err := c.ValidateChannel(chain, height, pe); err != nil {
+			return err
+		}
+	}
```
Comment (anilcse) [inline cmd/config.go]: I think if the `ClientID` is nil, there's no point in validating connection or channel details. We can just return `err` if the ClientID is nil and connectionID is not nil. Similarly for ChannelID

```suggestion
	if pe.ClientID != "" {
		if err := c.ValidateClient(chain, height, pe); err != nil {
			return err
		}
		
		if pe.ConnectionID != "" {
		      if err := c.ValidateConnection(chain, height, pe); err != nil {
			      return err
		      }
		      
		      if pe.ChannelID != "" {
		            if err := c.ValidateChannel(chain, height, pe); err != nil {
			            return err
		            }
	            }
	      }
	}
	
	if pe.ClientID == "" && pe.ConnectionID != "" {
	    return fmt.Errorf("ClientID is not conifgured for the connection: %s", pe.ConnectionID)
	}
```

**Review by anilcse — COMMENTED** [review]
lgtm, just small nits

**Review by anilcse — APPROVED** [review]

**Pull request merged**